commerce_coordinator/apps/titan/pipeline.py

```diff
@@ -140,3 +144,25 @@ def run_filter(self, edx_lms_user_id):  # pylint: disable=arguments-differ
             'order_data': order_data,
             'recent_payment': recent_payment
         }
+
+
+class UpdateBillingAddress(PipelineStep):
+    """
+    Updates the user's billing address info on an order in Titan
+    """
+
+    def run_filter(self, order_uuid, **kwargs):  # pylint: disable=arguments-differ
+
+        api_client = TitanAPIClient()
+        try:
+            response = api_client.update_billing_address(order_uuid, **kwargs)
+
+        except HTTPError as exc:
+            logger.exception(
+                "[UpdateBillingAddress] Failed to update the billing address for the specified order: %s",
+                order_uuid
+            )
+            raise APIException("Error updating the order's billing address details in titan") from exc
+        update_billing_address_output = BillingAddressSerializer(data=response)
+        update_billing_address_output.is_valid(raise_exception=True)
+        return update_billing_address_output.data
```

Comment:
Pipeline steps should return info in wrapped form. for example, here we can return 
```
{ 'billing_address_data': update_billing_address_output.data}
```
This helps to wire these steps in series and pass data to the next step in good shape.

Reply:
i think this is fair, especially as we try to funnel data through multiple steps... Should we also merge this with the input so the next step has everything?

Reply:
@grmartin Yes, good point. about input, the filter passes inputs to every step in the pipeline, so we do not need to merge input with output.

Reply:
updated to wrap return data